x/src/cargo.rs

```diff
@@ -106,6 +108,32 @@ impl Cargo {
         self
     }
 
+    /// Passes extra environment variables to x's target command.
+    pub fn envs<I, K, V>(&mut self, vars: I) -> &mut Self
+    where
+        I: IntoIterator<Item = (K, V)>,
+        K: AsRef<OsStr>,
+        V: AsRef<OsStr>,
+    {
+        for tup in vars {
```

Comment:
No need to change this, but more as an FYI you can use patterns even in for loops:
```suggestion
        for (key, value) in vars {
```

Reply:
I'll try again, pretty sure I ran in to compiler problems in the IDE.   It may have been more of a rust analyzer problem than a command line one.   I don't always force a command line build while experimenting.   It seemed for rust analyzer any info of the internals of the tuple's constituent type info were lost, and I had no way to force the compiler to accept it.